File: 0053_med_hot_maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length;

        if (n == 0) {
            return 0;
        }

        int f = 0;
        int result = Integer.MIN_VALUE;

        for (int num : nums) {
            if (f > 0) {
                f += num;
            } else {
                f = num;
            }
            result = Math.max(f, result);
        }

        return result;
    }
}

File: 0053_med_hot_maximum-subarray/solution.py
from typing import List


class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if not nums:
            return None
        
        f = 0
        ans = float('-inf')
        
        for i in range(len(nums)):
            if f > 0:
                f = f + nums[i]
            else:
                f = nums[i]
            ans = max(f, ans)
                
        return ans  